types/react-navigation/index.d.ts

```diff
@@ -1162,9 +1162,13 @@ export interface NavigationInjectedProps {
   navigation: NavigationScreenProp<NavigationState>;
 }
 
-export function withNavigation<T = {}>(
+export function withNavigation<
+  T = {},
+  C extends React.Component<T & NavigationInjectedProps> =
```

Comment:
If there's a way to get the `React.Component` from the `React.ComponentType`, that would be better; that way, it could infer the specific `onRef` type from the type of `Component`. But I'm not sure if there's a way to do that, so if there is let me know and I'll do that instead!

Reply:
And if there's a way to get the type of the `Props` type parameter from the `React.ComponentType` as well, that would be autofillable as well. So even more reason for that. But I'm unaware how you'd do it.

Reply:
But I think the React typings are too general to allow for this kind of query to work properly; I was thinking maybe to try to lookup `C`'s constructor's return type in the case it is a `React.ComponentClass`, but a) I don't know how to use lookup types to get a constructor, and b) it just returns a `React.Component`, not a more specific instance, so that probably wouldn't work either. Meh, this is good enough.